# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Lowering.cpp
# Commit: a1ca52a54ce8
# Full Hash: a1ca52a54ce80493e975350f1cc00f4eca15ed76
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2023-01-27 09:46:52
# Description:
#   Bug 1811803 - Use useBoxFixedAtStart in LMegemorphicSetElement r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167870
# ==============================================================================

diff -r b27d83c0aa57 -r a1ca52a54ce8 js/src/jit/Lowering.cpp
--- a/js/src/jit/Lowering.cpp	Thu Jan 26 18:40:45 2023 +0000
+++ b/js/src/jit/Lowering.cpp	Thu Jan 26 18:46:08 2023 +0000
@@ -4968,8 +4968,10 @@
   // See comment in LIROps.yaml (x86 is short on registers)
 #ifdef JS_CODEGEN_X86
   auto* lir = new (alloc()) LMegamorphicSetElement(
-      useRegisterAtStart(ins->object()), useBoxAtStart(ins->index()),
-      useBoxAtStart(ins->value()), tempFixed(CallTempReg0));
+      useRegisterAtStart(ins->object()),
+      useBoxFixedAtStart(ins->index(), CallTempReg1, CallTempReg2),
+      useBoxFixedAtStart(ins->value(), CallTempReg3, CallTempReg4),
+      tempFixed(CallTempReg0));
 #else
   auto* lir = new (alloc()) LMegamorphicSetElement(
       useRegisterAtStart(ins->object()), useBoxAtStart(ins->index()),
